### PR TITLE
Use ConcurrentDictionary to store reference to Orchestrator

### LINKED ISSUE #5134: OrchestratorRecognizer: Fix ability to use different models in same bot
As pointed out by @carlosscastro (thanks Carlos), Orchestrator Recognizer has a bug that prevents using multiple language models in the same bot (the orchestrator field declared `static` and therefore first model initialized wins). This has to be fixed to allow usage for multi-lang scenarios, where multiple models might need to be specified in single bot for various languages.

## Expected behavior
Bot user should be able to specify and use multiple models in whichever configuration for Orchestrator:
- Orchestrator should be usable in a configuration where one model can be used for one dialog, and another completely different model for another dialog in the same bot
- Orchestrator should be usable when used in a `MultiLanguageRecognizer` set - Orchestrator will use a default English model specialized on English, and another model (or multiple models) for other languages.

### PR DESCRIPTION
**Pull request by taicchoumsft**
Fixes #5134 

## Description
`orchestrator` was originally using static storage.  When using different Orchestrator models for different dialogs (for instance, in multi-language or fine-tuned scenarios), this means the first set `orchestrator` object will not be reinitialized, and the same `LabelResolver`  might be re-used for all other dialogs. 

## Specific Changes
Use a `ConcurrentDictionary` to store references to the `orchestrator` object.  The dictionary key is the full file path to the model that is being used. Thanks @carlosscastro for spotting the problem and pairing up to fix this. 

## Testing
Tested with a bot created with Composer, using multiple Orchestrator recognizers (One model for parent dialog, another completely different model for child)

### CODE REVIEW

**Review by tsuwandy — APPROVED** [review]
@taicchoumsft, did you test with both en and multi lang models?  By model, you meant the pretrained model, not the snapshot file, correct?

**Comment by taicchoumsft** [comment]
> @taicchoumsft, did you test with both en and multi lang models? By model, you meant the pretrained model, not the snapshot file, correct?

Yes I tested with multilang unicoder model in one dialog, and the 6 layer en model in a child dialog (same bot), and vice versa, and the correct label resolver is used for each case.   Yes to the pre-trained models, the snapshot files load correctly for each dialog once this fix is in.

**Comment by carlosscastro** [comment]
@taicchoumsft the change looks good, thanks

@tsuwandy @taicchoumsft not as part of this PR, but we should have some way of abstracting an orchestrator model loading in a way that does not require a huge model. Then, have the recognizer optionally receive that abstraction and test this recognizer through unit tests. 

One easy way of doing without public surface area is something like this (many other ways of doing this, just throwing a quick option)

```csharp
public clas OrchestratorRecognizer
{
    // for real life, uses the default IOrchestratorModelBuilder implementation which contains all the logic for loading the model from a file
    public OrchestratorRecognizer() {}

    // for test, uses test implementations of model builder that don't require to load a huge model
    internal OrchestratorRecognizer(IOrchestratorModelBuilder builder) {}
}
```

The bottom line is that we should have tests for these multi model scenarios soon, specially if we are taking hard dependencies on orchestrator for R12 and R13 (which we are).

**Review by carlosscastro — APPROVED** [review]

**Comment by tsuwandy** [comment]
@carlosscastro, thanks for the feedback, I have created a task in our repo to track as soon as R12 is done.

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs [hunk 1]
```diff
@@ -231,35 +232,37 @@ private void InitializeModel()
...
             {
-                throw new ArgumentNullException($"Missing `ShapshotFile` information.");
+                throw new ArgumentNullException($"Missing `SnapshotFile` information.");
             }
 
-            if (orchestrator == null && _resolver == null)
+            if (_resolver != null)
             {
-                var fullModelPath = Path.GetFullPath(PathUtils.NormalizePath(_modelFolder));
+                return;
```
Comment (daveta) [inline libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs:240]: (Let me preface, I think this is a fringe scenario, feel free to ignore).
I wonder if in this case we should verify consistency with passed in snapshot/model?

**Review by daveta — APPROVED** [review]

**Review by hcyang — APPROVED** [review]

**Review by mrivera-ms — APPROVED** [review]
